pkg/trace/sampler/memory_backend.go

```diff
@@ -69,6 +69,15 @@ func (b *MemoryBackend) CountSignature(signature Signature) {
 	b.mu.Unlock()
 }
 
+// CountClientDropped counts client dropped traces. They are added
+// to the totalScore, allowing them to weight on sampling rates during
+// adjust calls
+func (b *MemoryBackend) CountClientDropped(dropped int64) {
```

Comment:
The one thing that (IMHO) makes the sampler code more complicated is the lack of separation of concerns. In this example, this should be a memory backend, living agnostically of all the places where it is used. In its context, there is no such thing as "client dropped", there is only a concept of "total score".

I don't know if it's worth doing it here, but in my opinion this should be called `AddTotalScore` and the memory backend described in terms of its internal components and not its outer ones (how and where it's used).

If that's a lot of work, no need to do it here, but I wanted to leave this idea for you in case you can think of improvements.